src/parser.rs

```diff
@@ -5372,11 +5373,57 @@ impl<'a> Parser<'a> {
                 name,
                 alias,
                 args,
+                columns_definition,
                 with_hints,
             })
         }
     }
 
+    fn parse_redshift_columns_definition_list(
+        &mut self,
+        name: &ObjectName,
+    ) -> Result<Option<ColsDefinition>, ParserError> {
+        if !dialect_of!(self is RedshiftSqlDialect) {
```

Comment:
These functions are specific for the Redshift: https://docs.aws.amazon.com/redshift/latest/dg/r_System_information_functions.html so I think it's not needed for GenericDialect.

Reply:
I know, but the Generic dialect is supposed to be the most permissive one. Unless there are syntax conflicts (e.g., accepting those would break another part of the code), I can't see why not.